### PR TITLE
Refactor Digital Signature- LRQA-83256| master

### PR DESCRIPTION
**Pull request by juliosilveira77**
https://liferay.atlassian.net/browse/LRQA-83256

### CODE REVIEW

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by juliosilveira77** [comment]
ci:test:sf

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 1]
```diff
@@ -34,188 +38,172 @@ definition {
...
+		}
 
-		AssertElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+		task ("When a message appears that the envelope was successfully created") {
+			Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
+		}
 
-		DigitalSignature.deleteEnvelope(envelopeName = "Poshi Test HaveMoreRecipientsThanOne");
+		task ("Then it shows that more than one recipient has been created") {
+			Refresh();
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:195]: Could you change this task to: `Then the user can see that  an envelope have more than one recipient on list`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 2]
```diff
@@ -34,188 +38,172 @@ definition {
...
 
-		WaitForElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+		task ("And add another recipient") {
+			DigitalSignature.addMoreRecipient(
+				fullName = "Recipient Full Name 2",
+				recipientEmail = "test2@liferay.com");
+		}
 
-		AssertElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+		task ("When a message appears that the envelope was successfully created") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:190]: You don't need this task, you can put this step to view sucess message in the previous task

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 3]
```diff
@@ -34,188 +38,172 @@ definition {
...
+				addEmail = "test@liferay.com",
+				addFullName = "Recipient Full Name",
+				addMessage = "Email Message",
+				addName = "Poshi Test HaveMoreRecipientsThanOne",
+				addSubject = "Email Subject",
+				dmDocumentFile = "Alice's Adventures in Wonderland.pdf");
+		}
 
-		WaitForElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+		task ("And add another recipient") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:184]: Please change this task to: `"When the user adds more than one recipient"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 4]
```diff
@@ -34,188 +38,172 @@ definition {
...
-		Type(
-			locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_EXTRA_RECIPIENT_EMAIL",
-			value1 = "test2@liferay.com");
-
-		Button.click(button = "Send");
-
-		Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
-
-		Refresh();
+		task ("Given a digital envelope is created") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:174]: Please change this task to: `"Given a digital envelope is created"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 5]
```diff
@@ -34,188 +38,172 @@ definition {
...
-		AssertElementPresent(
-			dsStatus = ${dsStatus},
-			locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_STATUS");
+		task ("Then the status is checked") {
+			DigitalSignature.viewDigitalSignaturestatus(viewStatus = "Sent");
+		}
 	}
 
-	@description = "This is a test for LPS-137169. Verify if have more recipients than one"
+	@description = "LPS-131206. Verify if have more recipients than one"
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:171]: Here you can change the test description to: `Verify if is possible view on list when an envelope have more recipients than one`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 6]
```diff
@@ -34,188 +38,172 @@ definition {
...
+				addName = "Poshi Test CanViewCorrectDate",
+				addSubject = "Email Subject",
+				dmDocumentFile = "Alice's Adventures in Wonderland.pdf");
+
+			Button.click(button = "Send");
+		}
 
-		Refresh();
+		task ("When a message appears that the envelope was successfully created") {
+			Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:119]: Here you can remove this task and add this step inside the first task When

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 7]
```diff
@@ -34,188 +38,172 @@ definition {
...
-			addFullName = "Recipient Full Name",
-			addMessage = "Email Message",
-			addName = "Poshi Test CanViewCorrectDate",
-			addSubject = "Email Subject",
-			dmDocumentFile = "Alice's Adventures in Wonderland.pdf");
-
-		Button.click(button = "Send");
-
-		Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
+		task ("Given a digital envelope is created") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:106]: Change this macro to `"When a digital envelope is created"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 8]
```diff
@@ -34,188 +38,172 @@ definition {
...
 	@priority = 3
 	test CanDownloadEnvelopeByEnvelopeList {
-		DigitalSignature.enableDigitalSignature(siteStrategy = "Always Inherit");
-
-		Navigator.openURL();
-
-		ProductMenu.gotoPortlet(
-			category = "Content & Data",
-			portlet = "Digital Signature");
+		task ("When a User Test Goes to Digital Signature") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:74]: Please change this task to: `"When a User Test Goes to Envelope List"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 9]
```diff
@@ -34,188 +38,172 @@ definition {
...
 
-		Click(
-			envelopeName = ${envelopeName},
-			locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_ENVELOPE_NAME");
+			ProductMenu.gotoPortlet(
+				category = "Content & Data",
+				portlet = "Digital Signature");
 
-		AssertElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_DOWNLOAD_BUTTON");
+			DigitalSignature.enabledDigitalSignatureDownload(documentName = "Test");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:67]: This macro `DigitalSignature.enabledDigitalSignatureDownload` you can put in a task: `"Then the digital signature's envelope  can be downloaded"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 10]
```diff
@@ -34,188 +38,172 @@ definition {
...
 
-		ProductMenu.gotoPortlet(
-			category = "Content & Data",
-			portlet = "Digital Signature");
+		task ("Then a message appears that the envelope was successfully created") {
+			Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
+		}
 
-		var envelopeName = "Test";
+		task ("Then the digital signature can be downloaded") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:60]: Please change this task to: `"When the user goes to Envelope Description"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 11]
```diff
@@ -34,188 +38,172 @@ definition {
...
+
+			Button.click(button = "Send");
+		}
 
-		ProductMenu.gotoPortlet(
-			category = "Content & Data",
-			portlet = "Digital Signature");
+		task ("Then a message appears that the envelope was successfully created") {
+			Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
+		}
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:58]: Here you can remove this task and add this step inside the first task Given

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase [hunk 12]
```diff
@@ -34,188 +38,172 @@ definition {
...
-			addName = "Test",
-			addSubject = "Email Subject",
-			dmDocumentFile = "Alice's Adventures in Wonderland.pdf");
-
-		Button.click(button = "Send");
-
-		Alert.viewSuccessMessageText(successMessage = "Your envelope was created successfully.");
-
-		Navigator.openURL();
+		task ("Given a digital envelope is created") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:44]: Could you change this task to: "`Given a digital signature envelope is created"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 13]
```diff
@@ -340,12 +366,50 @@ definition {
...
+	}
+
+	macro viewMoreRecipient {
+		WaitForElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+
+		AssertElementPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
+
+		DigitalSignature.deleteEnvelope(envelopeName = ${documentName});
+
+		AssertElementNotPresent(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_RECIPIENTS_BADGE");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:410]: Could you remove these steps: `DigitalSignature.deleteEnvelope` and` AssertElementNotPresent` due to  these stepes don't belong to the view recipient process? You don't need create another macro for these ones

**Review by MGabriellaS — COMMENTED** [review]
@juliosilveira77  Thank you for this job, I request just some little things here. Could you take a look?

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase (hunk 4 above)
Comment (juliosilveira77) [reply modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:174]: Sua sugestao ta o mesmo ja no teste

Comment (MGabriellaS) [reply modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureListView.testcase:174]: Sorry @juliosilveira77  change to: `"Given a digital envelope is in creation"`

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by MGabriellaS** [comment]
@juliosilveira77 The code looks good to me!

**Comment by MGabriellaS** [comment]
ci:forward

**Pull request closed without merging**